addons/web/models/ir_qweb.py

```diff
@@ -45,17 +46,26 @@ def record_to_html(self, record, field_name, options):
         sha = hashlib.sha1(str(getattr(record, '__last_update')).encode('utf-8')).hexdigest()[0:7]
         max_size = '' if max_size is None else '/%s' % max_size
         avoid_if_small = '&avoid_if_small=true' if options.get('avoid_if_small') else ''
-        src = '/web/image/%s/%s/%s%s?unique=%s%s' % (record._name, record.id, options.get('preview_image', field_name), max_size, sha, avoid_if_small)
 
-        alt = None
+        if options.get('filename-field') and getattr(record, options['filename-field'], None):
+            filename = record[options['filename-field']]
```

Comment:
can we have smarter code, if options.get('filename') in record: record[options['filename']] else options.get('filename') or record.display_name

Reply:
it will void to have n^ option to do ~ the same thing ~

Reply:
I was thinking about it too, but it's not done like this for `alt`. Maybe it needs to be changed too then?

If we do it like that, we can't give a value that would be equal to a field name... Maybe it's important, maybe not, I'm not sure.